proposals/stdlib/occurrences-of.md

```diff
@@ -0,0 +1,57 @@
+# Extensions to find all occurrences of a pattern in a CharSequence
+
+* **Type**: Standard Library API proposal
+* **Author**: Christian Brüggemann
+* **Status**: Submitted
+* **Prototype**: [Implemented](https://github.com/JetBrains/kotlin/pull/821)
+
+
+## Summary
+
+An extension method `CharSequence.occurrencesOf(pattern: CharSequence, ignoreCase: Boolean, matchOverlapping: Boolean): Sequence<Int>` is proposed. 
+It returns a lazy sequence of all indices where the non-regex pattern is found in the receiver, based on a linear-time algorithm.
+If `ignoreCase` is set to `true`, the method treats all characters case-insensitively, meaning that two characters are considered equal if `Char.equals(other = x, ignoreCase = true)` returns true.
+If `matchOverlapping` is set to `false`, only non-overlapping occurrences are matched, meaning that when an occurrence is found, the method does not look for occurrences of the pattern straddling the occurrence that was found just now. Otherwise, they are matched as well.
+
+## Similar API review
+
+* Repeated calls to `indexOf` with incremented startIndex with superlinear runtime, especially if overlapping occurrences are searched for.
```

Comment:
I believe `indexOf` is not the similar API, because it returns an index of a single occurence, rather than indices of all.